examples/persistence_mongodb.ipynb

```diff
@@ -0,0 +1,490 @@
+    "        if config[\"configurable\"].get(\"thread_ts\"):\n",
+    "            doc[\"parent_ts\"] = config[\"configurable\"][\"thread_ts\"]\n",
+    "        self.collection.insert_one(doc)\n",
+    "        return {\n",
+    "            \"configurable\": {\n",
+    "                \"thread_id\": config[\"configurable\"][\"thread_id\"],\n",
+    "                \"thread_ts\": checkpoint[\"id\"],\n",
+    "            }\n",
+    "        }"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## MongoDB connection"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "MONGO_URI = \"mongodb://localhost:27017/\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Basic example using graph"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "StateSnapshot(values=4, next=(), config={'configurable': {'thread_id': '123'}}, metadata={'source': 'loop', 'step': 1, 'writes': {'add_one': 4}}, created_at='2024-07-09T05:59:02.742950+00:00', parent_config={'configurable': {'thread_id': '123', 'thread_ts': '1ef3db85-7c1c-69d5-8000-fa9878e4f2b9'}})"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from langgraph.graph import StateGraph\n",
+    "\n",
+    "checkpointer = MongoDBSaver(MongoClient(MONGO_URI), \"checkpoints_db\", \"checkpoints_collection\")\n",
+    "builder = StateGraph(int)\n",
+    "builder.add_node(\"add_one\", lambda x: x + 1)\n",
+    "builder.set_entry_point(\"add_one\")\n",
+    "builder.set_finish_point(\"add_one\")\n",
+    "graph = builder.compile(checkpointer=checkpointer)\n",
+    "config = {\"configurable\": {\"thread_id\": \"123\"}}\n",
+    "graph.get_state(config)\n",
+    "result = graph.invoke(3,config)\n",
+    "graph.get_state(config)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "4"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "result"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup environment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import getpass\n",
+    "import os\n",
+    "\n",
+    "\n",
+    "def _set_env(var: str):\n",
+    "    if not os.environ.get(var):\n",
+    "        os.environ[var] = getpass.getpass(f\"{var}: \")\n",
+    "\n",
+    "\n",
+    "_set_env(\"OPENAI_API_KEY\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup model and tools for the graph"
```

Comment:
are you using this at all in the notebook? it seems disconnected from the basic usage example